source/material_model/rheology/strain_dependent.cc

```diff
@@ -450,6 +469,41 @@ namespace aspect
 
       }
 
+      template <int dim>
+      std::array<double, 3>
+      StrainDependent<dim>::
+      apply_temperature_dependence_to_strain_weakening_factors(const unsigned int j,
+                                                               const double temperature,
+                                                               std::array<double, 3> weakening_factors) const
+      {
+
+        //
+        //  ------            -------- 1
+        //        \          /
+        //         \        /
+        //          \______/ _ _ _ _ _ viscous_weakening_factor[2]
+        //
+        //  ----------------------------> T
+        //      T0  T1   T2  T3
+
+        double factor=1;
+        const double T0=viscous_strain_weakening_T0[j];
+        const double T1=viscous_strain_weakening_T1[j];
+        const double T2=viscous_strain_weakening_T2[j];
+        const double T3=viscous_strain_weakening_T3[j];
+
+        if (temperature>T0 && temperature<=T1)
+          factor=(weakening_factors[2]-1.)/(T1-T0)*(temperature-T0)+1;
```

Comment:
Personal preference, but:
```suggestion
          factor = (weakening_factors[2]-1.) / (T1-T0)*(temperature-T0) + 1.;
```